File: cw/cpp/fake_binary.cpp
// https://www.codewars.com/kata/57eae65a4321032ce000002d/train/cpp


#include <iostream>

std::string fakeBin(std::string str) {
    for(int i = 0; i < str.size(); ++i) {
        int n = str[i] - '0';
        if(n < 5) {
            str.replace(i, 1, "0");
        } else {
            str.replace(i, 1, "1");
        }
    }

    return str;
}

int main() {
    std::string str;

    std::cin >> str;
    std::cout << fakeBin(str) << "\n";

    return 0;
}

File: codewars/perl/fake_binary.pl
# https://www.codewars.com/kata/57eae65a4321032ce000002d/train/perl

package Solution;

use 5.30.0;
use strict;
use warnings;
use Exporter qw(import);
our @EXPORT_OK = qw(comp);

sub fake_bin {
    my $input = $_[0];
    my $bin = "";

    foreach my $char (split('', $input)) {
        if($char lt 5) {
            $bin = $bin."0";
        } else {
            $bin = $bin."1";
        }
    }

    return $bin
}

my $input = <>;

print(fake_bin($input))
